adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala

```diff
@@ -888,7 +1310,80 @@ private[adam] class VariantContextConverter(
     }
   }
 
-  private[converters] def makeHtsjdkGenotypeConverter(
+  def makeVariantFormatFn(
+    headerLines: Seq[VCFHeaderLine]): (HtsjdkVariantContext, Option[String], Int) => Variant = {
+
+    val attributeFns: Iterable[(HtsjdkVariantContext, Int, Array[Int]) => Option[(String, String)]] = headerLines
+      .flatMap(hl => hl match {
+        case il: VCFInfoHeaderLine => {
+          // get the id of this line
+          val key = il.getID
+
+          // filter out the lines that we already support
+          if (SupportedHeaderLines.infoHeaderLines
+            .find(_.getID == key)
+            .isEmpty) {
+
+            Some(lineToVariantContextExtractor(il))
+          } else {
+            None
+          }
+        }
+        case _ => None
+      })
+
+    def convert(vc: HtsjdkVariantContext,
+                alt: Option[String],
+                alleleIdx: Int): Variant = {
+
+      // create the builder
+      val variantBuilder = Variant.newBuilder
+        .setContigName(vc.getChr)
+        .setStart(vc.getStart - 1)
+        .setEnd(vc.getEnd)
+        .setReferenceAllele(vc.getReference.getBaseString)
+
+      alt.foreach(variantBuilder.setAlternateAllele(_))
+
+      // bind the conversion functions and fold
+      val boundFns: Iterable[Variant.Builder => Variant.Builder] = variantFormatFns
+        .map(fn => {
+          fn(vc, _: Variant.Builder)
+        })
+      val converted = boundFns.foldLeft(variantBuilder)((vb: Variant.Builder, fn) => fn(vb))
+
+      val variant = variantBuilder.build
+      val variantAnnotationBuilder = VariantAnnotation.newBuilder
+
+      val boundAnnotationFns: Iterable[VariantAnnotation.Builder => VariantAnnotation.Builder] = variantAnnotationFormatFns
+        .map(fn => {
+          fn(vc, _: VariantAnnotation.Builder, variant, alleleIdx)
+        })
+      val convertedAnnotation = boundAnnotationFns.foldLeft(variantAnnotationBuilder)(
+        (vab: VariantAnnotation.Builder, fn) => fn(vab))
+
+      // todo: I don't believe these are useful for INFO fields
+      val indices = Array.empty[Int]
+
+      // pull out the attribute map and process
+      val attrMap = attributeFns.flatMap(fn => fn(vc, alleleIdx, indices))
+        .toMap
+
+      // if the map has kv pairs, attach it
+      val convertedAnnotationWithAttrs = if (attrMap.isEmpty) {
+        convertedAnnotation
+      } else {
+        convertedAnnotation.setAttributes(attrMap)
+      }
+
+      variant.setAnnotation(convertedAnnotationWithAttrs.build)
+      variant
```

Comment:
Instead, I would call `variantBuilder.build` here, after you attach the annotation.